src/node/internal/internal_http_client.ts

```diff
@@ -0,0 +1,372 @@
+    }
+
+    if (options.joinDuplicateHeaders !== undefined) {
+      validateBoolean(
+        options.joinDuplicateHeaders,
+        'options.joinDuplicateHeaders'
+      );
+    }
+
+    this.joinDuplicateHeaders = Boolean(options.joinDuplicateHeaders);
+
+    this.path = options.path || '/';
+    if (cb) {
+      this.once('response', cb);
+    }
+
+    this.host = host;
+
+    const headers = options.headers;
+    if (!Array.isArray(headers)) {
+      if (headers != null) {
+        for (const [key, value] of Object.entries(headers)) {
+          this.setHeader(key, value as unknown as string);
+        }
+      }
+
+      if (host && !this.getHeader('host') && setHost) {
+        let hostHeader = host;
+
+        // For the Host header, ensure that IPv6 addresses are enclosed
+        // in square brackets, as defined by URI formatting
+        // https://tools.ietf.org/html/rfc3986#section-3.2.2
+        const posColon = hostHeader.indexOf(':');
+        if (
+          posColon !== -1 &&
+          hostHeader.includes(':', posColon + 1) &&
+          hostHeader.charCodeAt(0) !== 91 /* '[' */
+        ) {
+          hostHeader = `[${hostHeader}]`;
+        }
+
+        if (port && +port !== defaultPort) {
+          // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+          hostHeader += ':' + port;
+        }
+        this.setHeader('Host', hostHeader);
+      }
+
+      if (options.auth && !this.getHeader('Authorization')) {
+        this.setHeader(
+          'Authorization',
+          'Basic ' + Buffer.from(options.auth).toString('base64')
+        );
+      }
+    }
+
+    this.on('finish', () => {
+      this.#onFinish();
+    });
+  }
+
+  public setHeader(name: string, value?: string): void {
+    validateHeaderName(name);
+    validateHeaderValue(name, value);
+
+    // Node.js uses an array, but we store it as an object.
+    this[kOutHeaders][name.toLowerCase()] = {
+      name,
+      value: value as string,
+    };
+  }
+
+  public getHeader(name: string): string | undefined {
+    return this[kOutHeaders][name.toLowerCase()]?.value;
+  }
+
+  public removeHeader(name: string): void {
+    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+    delete this[kOutHeaders][name.toLowerCase()];
+  }
+
+  #onFinish(): void {
+    if (this.destroyed) return;
+
+    let body: BodyInit | null = null;
+    if (this.method !== 'GET' && this.method !== 'HEAD') {
+      body = new Blob(this.#body, {
+        type: this.getHeader('content-type') ?? '',
+      });
+    }
+
+    const headers: [string, string][] = [];
+    for (const name of Object.keys(this)) {
+      const value = this[kOutHeaders][name]?.value;
+      if (value) {
+        if (Array.isArray(value)) {
+          for (const item of value) {
+            headers.push([name, item as string]);
+          }
+        } else {
+          headers.push([name, value]);
+        }
+      }
+    }
+
+    const header = this.getHeader('host');
```

Comment:
```suggestion
    const host = this.getHeader('host');
```

I was really confused when I saw `http://${header}` later.